src/dft/backends/cufft/execute_helper.hpp

```diff
@@ -143,6 +147,27 @@ inline CUstream setup_stream(const std::string &func, sycl::interop_handle ih, c
     return stream;
 }
 
+
+/** Wrap interop API to launch interop host task.
+ * 
+ * @tparam HandlerT The command group handler type
+ * @tparam FnT The body of the enqueued task
+ *
+ * Either uses host task interop API, or enqueue native command extension.
+ * This extension avoids host synchronization after 
+ * the CUDA call is complete.
+ */
+template <typename HandlerT, typename FnT>
+static inline void cufft_enqueue_task(HandlerT&& cgh, FnT&& f) {
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle ih){
+#else
+    cgh.host_task([=](sycl::interop_handle ih){
+#endif
+        f(std::move(ih));
```

Comment:
Why is it needed to duplicate this in both cuFFT and rocFFT backends and in various domains (BLAS, LAPACK, FFT)? Can't we have one wrapper used across all domains and backends?

Reply:
My understanding is that the domains have always been separated on purpose. This makes review much easier as any change affecting common code would technically require an approval from every domain owners.
I agree this could be discussed in an issue. To my knowledge there is very little code that could be common across domains, other than the types and exceptions which are already common.